app/controllers/search_controller.rb

```diff
@@ -0,0 +1,5 @@
+class SearchController < ApplicationController
+  def new
+    @results = Resource.search(params[:query])
```

Comment:
Strictly speaking, we only need SP's for update/create/delete actions, but good eye.